modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/DataSet.js

```diff
@@ -288,6 +299,17 @@ const DataSet = ({
 			});
 	}
 
+	useEffect(() => {
+		setSelectedItems(
+			getSelectedItems({
+				items,
+				key: selectedItemsKey,
+				values: selectedItemsValue,
+			})
+		);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [selectedItemsValue]);
```

Comment:
It is a bit unfortunate to have these parallel states with partially duplicated values. Ideally, we would have only `selectedItems`, as `selectedItemsValues` is a subset of it. Then, if there is a need, we could a utility like `getSelectedItemsValues`, that would extract only `id`s from `selectedItems`. This might be a pretty extensive change though :thinking:

Reply:
Yeah! I thought the same way, it's not nice to have this overlapping values, group them in an object seems a nice solution. However, for now, I'll just keep it this way, trying not to break things. Do you agree?